S2VX.Game.Tests/VisualTests/S2VXCursorTests/CursorReset.cs

```diff
@@ -0,0 +1,36 @@
+﻿using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Screens;
+using S2VX.Game.SongSelection;
+
+namespace S2VX.Game.Tests.VisualTests.S2VXCursorTests {
+    public class CursorReset : S2VXTestScene {
+        [Cached]
+        private ScreenStack ScreenStack { get; set; } = new();
+
+        private SongSelectionScreen SongSelectionScreen { get; set; }
+
+        [Cached]
+        private S2VXCursor Cursor { get; set; } = new S2VXCursor();
+
+        [BackgroundDependencyLoader]
+        private void Load() {
+            ScreenStack.Push(SongSelectionScreen = new());
+            Add(ScreenStack);
+            Add(Cursor);
+        }
+
+        [Test]
+        public void Reset_SongSelectionEntering_ResetsCursorProperties() {
+            AddStep("Update cursor rotation", () => Cursor.ActiveCursor.Rotation = 1);
+
+            // Instead of calling OnEntering manually, I tried doing
+            // ScreenStack.Push(new SongSelectionScreen). But this was
+            // unreliable in whether it would actually call
+            // SongSelectionScreen's OnEntering or not.
```

Comment:
It would call `OnEntering`, but not all the time. It seems like it's dependent on osu!framework's internals. Sometimes when you ran the test, `OnEntering` would be called, other times it wouldn't.

Reply:
I see this test has changed to test `OnExiting` instead. I assume that it's the same situation where if you call `ScreenStack.Exit()` it won't call `OnExiting` consistently?

Reply:
Also I ran this test in the GUI and I don't see any change in the screen (seems to just stay in a PlayScreen) or cursor rotation. Is this supposed to be a HeadlessTest?

Reply:
> I see this test has changed to test `OnExiting` instead. I assume that it's the same situation where if you call `ScreenStack.Exit()` it won't call `OnExiting` consistently?
Yep, same situation.

> Also I ran this test in the GUI and I don't see any change in the screen (seems to just stay in a PlayScreen) or cursor rotation. Is this supposed to be a HeadlessTest?
Yep, moved and changed to use `[HeadlessTest]`.